File: 49-Group-Anagrams/49.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String> > res = new HashMap<>();
        for(int i = 0; i < strs.length; i++){
            int[] count = new int[26];
            for(int j = 0; j < strs[i].length(); j++){
                count[strs[i].charAt(j) - 'a']++;
            }
            String key = Arrays.toString(count);
            res.putIfAbsent(key, new ArrayList<>());
            res.get(key).add(strs[i]);
        }
        return new ArrayList<>(res.values());
    }
}

File: 49-Group-Anagrams/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map <string, vector<string> > res;
        for(int i = 0; i < strs.size(); i++){
            vector<int> count(26,0);
            for(int j = 0; j < strs[i].size(); j++){
                count[strs[i][j] - 'a']++;
            }
            string key = to_string(count[0]);
            for(int j = 1; j < 26; j++){
                key += ',' + to_string(count[j]);
            }
            res[key].push_back(strs[i]);
        }
        vector<vector<string> > ans;
        for(auto& pair : res){
            ans.push_back(pair.second);
        }
        return ans;
    }
};